### ISSUE TITLE
NPE when Exec.getLogger is called

### ISSUE THREAD

**Issue by muga** [embulk/embulk-executor-mapreduce#9]
Because Exec.getLogger is called before Exec.doWith is executed.

``` java
java.lang.NullPointerException: Exec is used outside of Exec.doWith
    at org.embulk.spi.Exec.session(Exec.java:37)
    at org.embulk.spi.Exec.getLogger(Exec.java:59)
    at org.embulk.executor.mapreduce.EmbulkMapReduce.hadoopOperationWithRetry(EmbulkMapReduce.java:327)
    at org.embulk.executor.mapreduce.EmbulkMapReduce.readPluginArchive(EmbulkMapReduce.java:235)
    at org.embulk.executor.mapreduce.EmbulkMapReduce$SessionRunner.readPluginArchive(EmbulkMapReduce.java:396)
    at org.embulk.executor.mapreduce.EmbulkPartitioningMapReduce$EmbulkPartitioningMapper.setup(EmbulkPartitioningMapReduce.java:52)
    at org.apache.hadoop.mapreduce.Mapper.run(Mapper.java:142)
    at org.apache.hadoop.mapred.MapTask.runNewMapper(MapTask.java:764)
    at org.apache.hadoop.mapred.MapTask.run(MapTask.java:340)
    at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:168)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.security.auth.Subject.doAs(Subject.java:415)
    at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1594)
    at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:163)
```

**Comment by frsyuki** [comment]
Thanks! Fixed at v0.1.4.

**Comment by muga** [comment]
I still hit NPE.
https://github.com/embulk/embulk-executor-mapreduce/blob/master/embulk-executor-mapreduce/src/main/java/org/embulk/executor/mapreduce/EmbulkPartitioningMapReduce.java#L159

``` java
java.lang.NullPointerException: Exec is used outside of Exec.doWith
 at org.embulk.spi.Exec.session(Exec.java:37) at org.embulk.spi.Exec.getLogger(Exec.java:59)
 at org.embulk.executor.mapreduce.EmbulkMapReduce.hadoopOperationWithRetry(EmbulkMapReduce.java:389)
 at org.embulk.executor.mapreduce.EmbulkMapReduce.readPluginArchive(EmbulkMapReduce.java:250)
 at org.embulk.executor.mapreduce.EmbulkMapReduce$SessionRunner.readPluginArchive(EmbulkMapReduce.java:463)
 at org.embulk.executor.mapreduce.EmbulkPartitioningMapReduce$EmbulkPartitioningReducer.setup(EmbulkPartitioningMapReduce.java:159)
 at org.apache.hadoop.mapreduce.Reducer.run(Reducer.java:168)
 at org.apache.hadoop.mapred.ReduceTask.runNewReducer(ReduceTask.java:627)
 at org.apache.hadoop.mapred.ReduceTask.run(ReduceTask.java:389)
 at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:168)
 at java.security.AccessController.doPrivileged(Native Method)
 at javax.security.auth.Subject.doAs(Subject.java:415)
 at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1594)
 at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:163) 
```
